File: tests/constructor.cc

class Foo {
 
public:
    Foo() {}
    Foo(int x) { a = x;}

    int bar();
    static int foo(int x, double y);

    double def(int x, double y = 2.0, char *str = "xyz");
protected:
    int a;
};

File: tests/constructor.R
library(RCodeGen)
library(RCIndex)

classes = getCppClasses("constructor.cc")

k = classes[[1]]
r = k@methods[[1]]

code = createCppMethod(r)
code

foo = createRProxy(k@methods$foo)

bar = createRProxy(k@methods$bar)

def = createRProxy(k@methods$def)



